### PR TITLE
Create /usr/local/bin if it doesn't exists when install cli

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jwcesign <jiangwei115@huawei.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Create /usr/local/bin if it doesn't exists when install cli

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/3106

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @dmulizhihui  @XiShanYongYe-Chang @RainbowMango

File: hack/install-cli.sh [hunk 1]
```diff
@@ -237,6 +237,15 @@ setup_binary() {
   info "Installing ${INSTALL_CLI_TYPE} to ${INSTALL_LOCATION}/${INSTALL_CLI_TYPE}"
   tar -xzof "${TMP_BIN}" -C "${TMP_DIR}" --no-same-owner
 
+  # in some production environments, directory /usr/local/bin may not exist
```
Comment (RainbowMango) [inline hack/install-cli.sh:240]: This comment is probably inappropriate, what we should do is `ensure the path of $INSTALL_LOCATION exist`, this is a configurable environment and not limited to `/usr/local/bin`.

File: hack/install-cli.sh [hunk 2]
```diff
@@ -237,6 +237,14 @@ setup_binary() {
   info "Installing ${INSTALL_CLI_TYPE} to ${INSTALL_LOCATION}/${INSTALL_CLI_TYPE}"
   tar -xzof "${TMP_BIN}" -C "${TMP_DIR}" --no-same-owner
 
+  if [[ ! -d "${INSTALL_LOCATION}" ]]; then
+    if [[ -w "${INSTALL_LOCATION}" ]]; then
```
Comment (RainbowMango) [inline hack/install-cli.sh:241]: Does this really work? Can we test a `non-exist` path writable?

**Comment by RainbowMango** [comment]
/assign @ikaven1024 
@ikaven1024 must be the expert.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Please help update the release note. Do we need to cherry-pick this patch to the previous release?

**Comment by RainbowMango** [comment]
Update release note to `NONE`?

**Comment by jwcesign** [comment]
We don't need to cherry-pick(it will not occur in general using). 
And also, it doesn't need release notes as it's not a bug of karmada's component.

**Pull request merged**